File: app/src/main/java/com/noticias_now/app/ApsNoticiasApp.java
package com.noticias_now.app;

import android.app.Activity;
import android.app.Application;
import android.text.TextUtils;

import com.noticias_now.model.UsuarioModel;
import com.noticias_now.util.ApsNoticiasUtils;

/**
 * Created by ricarlo on 10/11/2016.
 */

public class ApsNoticiasApp extends Application {

    private static ApsNoticiasApp instance;
    private UsuarioModel usuario;
    public static Activity currentActivity;

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
        verifyExistUser();
    }

    public static ApsNoticiasApp getInstance(){
        return instance;
    }

    public UsuarioModel getUsuario(){
        return usuario;
    }

    public void verifyExistUser(){
        String jsonUser = ApsNoticiasUtils.read(instance, Constants.SharedPreferences.PATH_APP, Constants.SharedPreferences.DATA_USER, null);
        if (!TextUtils.isEmpty(jsonUser)){
            usuario = (UsuarioModel) ApsNoticiasUtils.jsonToObject(jsonUser, UsuarioModel.class);
        }
    }

    public boolean checkIfLoggedin() {
        return usuario != null;
    }

    public void saveUser(UsuarioModel user){
        if (user != null){
            ApsNoticiasUtils.write(instance, Constants.SharedPreferences.PATH_APP, Constants.SharedPreferences.DATA_USER, ApsNoticiasUtils.objectToString(user));
            usuario = user;
        }
    }

    public void doLogout() {
        usuario = null;
        saveUserInfo("");
        ApsNoticiasUtils.write(this, Constants.SharedPreferences.PATH_APP, Constants.SharedPreferences.DEVICE_TOKEN, "");
        ApsNoticiasUtils.remove(this, Constants.SharedPreferences.PATH_APP, Constants.SharedPreferences.DEVICE_TOKEN);

    }

    private static void saveUserInfo(String userJSON) {
        ApsNoticiasUtils.write(instance, Constants.SharedPreferences.PATH_APP, Constants.SharedPreferences.DATA_USER, userJSON);
    }
}

File: app/src/main/java/com/noticias_now/app/ApsNoticiasApp.kt
package com.noticias_now.app

import android.app.Application
import android.text.TextUtils
import br.com.ricarlo.common.di.CommonModule
import com.noticias_now.model.UsuarioModel
import com.noticias_now.util.ApsNoticiasUtils
import org.koin.android.ext.koin.androidContext
import org.koin.core.context.startKoin

/**
 * Created by ricarlo on 10/11/2016.
 */
class ApsNoticiasApp : Application() {

    var usuario: UsuarioModel? = null
        private set

    override fun onCreate() {
        super.onCreate()
        startDI()
        instance = this
        verifyExistUser()
    }

    private fun verifyExistUser() {
        val jsonUser = ApsNoticiasUtils.read(instance, Constants.SharedPreferences.PATH_APP, Constants.SharedPreferences.DATA_USER, null)
        if (!TextUtils.isEmpty(jsonUser)) {
            usuario = ApsNoticiasUtils.jsonToObject(jsonUser, UsuarioModel::class.java) as UsuarioModel
        }
    }

    fun checkIfLogged(): Boolean {
        return usuario != null
    }

    fun saveUser(user: UsuarioModel?) {
        if (user != null) {
            ApsNoticiasUtils.write(instance, Constants.SharedPreferences.PATH_APP, Constants.SharedPreferences.DATA_USER, ApsNoticiasUtils.objectToString(user))
            usuario = user
        }
    }

    fun doLogout() {
        usuario = null
        saveUserInfo("")
        ApsNoticiasUtils.write(this, Constants.SharedPreferences.PATH_APP, Constants.SharedPreferences.DEVICE_TOKEN, "")
        ApsNoticiasUtils.remove(this, Constants.SharedPreferences.PATH_APP, Constants.SharedPreferences.DEVICE_TOKEN)
    }

    private fun startDI() {
        startKoin {
            // Android context
            androidContext(this@ApsNoticiasApp)
            // modules
            modules(CommonModule.modules)
        }
    }

    companion object {
        @JvmStatic
        var instance: ApsNoticiasApp? = null
            private set
        private fun saveUserInfo(userJSON: String) {
            ApsNoticiasUtils.write(instance, Constants.SharedPreferences.PATH_APP, Constants.SharedPreferences.DATA_USER, userJSON)
        }
    }
}